src/java.base/share/classes/java/net/Socket.java

```diff
@@ -554,6 +541,7 @@ void setImpl(SocketImpl si) {
      * Sets impl to the system-default type of SocketImpl.
      * @since 1.4
      */
+    @SuppressWarnings("deprecation")
     void setImpl() {
         SocketImplFactory factory = Socket.factory;
```

Comment:
The scope of `@SuppressWarnings` should be reduced as much as possible. I believe in this case it can be reduced to the assignation of the factory:

```
        @SuppressWarnings("deprecation")
        SocketImplFactory factory = Socket.factory;
```

Reply:
`@SuppressWarnings` moved down to next line as suggested. See e36f91d